plugins/woocommerce-blocks/assets/js/data/cart/utils.ts

```diff
@@ -116,3 +116,24 @@ export const validateDirtyProps = ( dirtyProps: {
 
 	return invalidProps.length === 0;
 };
+
+/**
+ * Gets the localStorage flag to indicate whether the customer data is dirty.
+ */
+export const getIsCustomerDataDirty = () => {
+	return (
+		window.localStorage.getItem(
+			'WOOCOMMERCE_CHECKOUT_IS_CUSTOMER_DATA_DIRTY'
+		) === 'true'
+	);
+};
+
+/**
+ * Sets a flag in localStorage to indicate whether the customer data has been modified.
+ */
+export const setIsCustomerDataDirty = ( isCustomerDataDirty: boolean ) => {
+	window.localStorage.setItem(
+		'WOOCOMMERCE_CHECKOUT_IS_CUSTOMER_DATA_DIRTY',
+		isCustomerDataDirty ? 'true' : 'false'
+	);
+};
```

Comment:
I'm still reviewing but I'm not sure how to best handle this? Different tabs can't know if local storage got updated unless you're calling localStorage each time which isn't that good, though we're reading booleans so it should be fine.

Do we need to clear this after an order is placed? is it okay for address to be marked dirty on future orders?

Reply:
> unless you're calling localStorage each time which isn't that good, though we're reading booleans so it should be fine.

I just pushed a change so that writing to local storage is debounced (300ms) and is no longer read each time the `SET_BILLING_ADDRESS` and `SET_SHIPPING_ADDRESS` is dispatched.

> Do we need to clear this after an order is placed? is it okay for address to be marked dirty on future orders?

Hmm, we should remove it from localStorage when the order is placed I guess, no need to keep it around any more. Added in 754a60bffbcc0d4b41a615dcb0b73ae4ec82ade9